src/middlewared/middlewared/plugins/smb_/groupmap.py

```diff
@@ -317,9 +315,11 @@ def groupmap_list(self):
                     continue
 
                 key = 'local'
-
-            if key == 'invalid' or rv[key].get(gid):
-                rv['invalid'].append(g['sid'])
+            else:
```

Comment:
Trying to grok the comment.  Do you mean:
`Found a groupmap that does not have a local machine sid to local group mapping.`

This is the catch-all clause at the end of a few `elif`. Why are 'all others' grouped into the `legacy entries` bucket?
The 'legacy' bucket holds 'unmapped'.  I guess I'm looking for some comments for what entries should fall into the 'legacy' bucket.

**_Also..._**
The docstring for this function needs to be updated as it refers to the now removed `invalid` entries.

Reply:
Updated docstring. Added better comments.